coins/ethereum/src/ethereum_wallet.rs

```diff
@@ -271,44 +297,170 @@ impl EthereumWallet {
+
+    // TODO: take chain_id as a parameter
+    // TODO: Take index as a parameter and use that for deriving the wallet we want (refactor keystore)
     /// Creates and sends a transfer transaction to the Ethereum blockchain.
-    pub async fn transfer(
+    async fn transfer(
         &self,
-        send_amount: EthereumAmount,
-        to_address: &str,
+        _send_amount: &Self::CryptoAmount,
+        _to_address: &str,
     ) -> Result<String, Error> {
-        let blockchain_client = self.blockchain_client()?;
-        let to = Address::from_str(to_address).map_err(|e| Error::FromStr(e.to_string()))?;
-        let amount = send_amount.wei();
-
-        let tx_object = TransactionParameters {
-            to: Some(to),
-            value: amount,
-            ..Default::default()
-        };
-
-        let secret_key = self.private_key()?.0;
 
-        // sign the tx
-        let signed = blockchain_client
-            .web3()
-            .accounts()
-            .sign_transaction(tx_object, &secret_key)
-            .await?;
+        let secret_key: &Result<EthereumPrivateKey, Error> = &self.private_key();
 
-        let result = blockchain_client
-            .eth()
-            .send_raw_transaction(signed.raw_transaction)
-            .await?;
-
-        let hash = hex::encode(result.as_bytes());
-
-        Ok(hash)
+        let derived_hd_key = &self.derived_hd_key()?;
+        let private_key =
+                EthereumPrivateKey::from_slice(&derived_hd_key.extended_private_key()?.to_bytes())?;
+        let _address_derivation_path = &derived_hd_key.derivation_path.clone();
+        
+        // EthereumWallet stores the private key as a 32 byte array
+        let secret_bytes = private_key.to_bytes();
+
+        // Instantiate wallet using the private key's secret bytes
+        let wallet_from_bytes = Wallet::from_bytes(&secret_bytes).unwrap();
+
+        let provider = &self.blockchain_client().unwrap().ethers();
+
+        // Instantiate a ethers local wallet from the wallet's secret bytes
+
+        // 5 = goerli chain id 
+
+        // Link our wallet instance to our provider for signing our transactions
+        let client = SignerMiddleware::new(provider, wallet_from_bytes.with_chain_id(5u64));    
+
+        // Create a transaction request to send 10000 wei to the Goerli address
+        let tx = TransactionRequest::new()
+            .to("0x681dA56258fF429026449F1435aE87e1B6e9F85b")
+            .gas(21000)
+            .value(10000)
+            .chain_id(5u64);
+
+
+        println!("Initialising send: {:?}", &tx);
+        
+        let pending_tx = client.send_transaction(tx, None).await.unwrap();
+        let receipt = pending_tx.await.unwrap().ok_or_else(|| println!("tx dropped from mempool")).unwrap();
+        
+        let _tx = client.get_transaction(receipt.transaction_hash).await.unwrap();
+        Ok("tx_id".to_string())
+        // let wallet: LocalWallet = LocalWallet::from(secret_key).with_chain_id(5);
```

Comment:
what is all the commented-out code needed for? there is no context to know why its needed